apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs

```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
+                };
+            InstanceProperties props = new InstanceProperties{
+                Disks = {
+                            new AttachedDisk{
+                                AutoDelete = true,
+                                Boot = true,
+                                Type = AttachedDisk.Types.Type.Persistent,
+                                InitializeParams = init_params
+                            }
+                        },
+                NetworkInterfaces = {
+                    new NetworkInterface{
+                        Name="default"
+                        }
+                },
+                MachineType = "n1-standard-1"
+            };
+            InstanceTemplate instanceTemplate = new InstanceTemplate
+            {
+                Name = instanceTemplateName,
+                Properties = props
+            };
+            try{
+                Operation insertTemplateOp = instanceTemplatesClient.Insert(
+                    projectId,
+                    instanceTemplate
+                );
+                PollForCompletionGlobalOp(insertTemplateOp, "insert template");
+
+                InstanceGroupManager instanceGroupManager = new InstanceGroupManager{
+                    InstanceTemplate = insertTemplateOp.TargetLink,
+                    TargetSize = 1,
+                    Name = instanceGroupManagerName,
+                    BaseInstanceName = "dotnetgapic"
+                };
+
+                Operation insertIgmOp = instanceGroupManagersClient.Insert(
+                    projectId,
+                    zone,
+                    instanceGroupManager
+                );
+                PollForCompletion(insertIgmOp, "insert group manager");
+
+                Operation resizeOp = instanceGroupManagersClient.Resize(
+                    projectId,
+                    zone,
+                    instanceGroupManagerName,
+                    0
+                );
+
+                PollForCompletion(resizeOp, "resize");
+
+                InstanceGroupManager instanceGroupManagerResized = instanceGroupManagersClient.Get(
+                    projectId,
+                    zone,
+                    instanceGroupManagerName
+                );
+
+                Assert.Equal(0, instanceGroupManagerResized.TargetSize);
+            }
+            finally{
+                Operation deleteOp = instanceGroupManagersClient.Delete(
+                    projectId,
+                    zone,
+                    instanceGroupManagerName
+                );
+
+                PollForCompletion(deleteOp, "delete igm");
+
+                Operation deleteTemplateOp = instanceTemplatesClient.Delete(
+                    projectId,
+                    instanceTemplateName
+                );
+
+                PollForCompletionGlobalOp(deleteTemplateOp, "delete template");
+            }
+
+        }
+
+        private Operation PollForCompletion(Operation operation, string alias)
```

Comment:
I really hope we don't need this code (and PollForCompletionGlobalOp) in multiple places.